crates/sui-core/src/authority/authority_store.rs

```diff
@@ -883,10 +883,28 @@ impl AuthorityStore {
         // Add batched writes for objects and locks.
         let effects_digest = effects.digest();
 
+        // Get the already-written markers for configs this epoch.
+        let mut already_written_config_updates = self
+            .perpetual_tables
+            .object_per_epoch_marker_table
+            .multi_contains_keys(
+                markers
+                    .iter()
+                    .filter(|(_, marker_type)| matches!(marker_type, MarkerValue::ConfigUpdate(_)))
+                    .map(|(k, _)| (epoch_id, *k))
+                    .rev(),
+            )?;
+
         write_batch.insert_batch(
             &self.perpetual_tables.object_per_epoch_marker_table,
             markers
                 .iter()
+                .filter(|(_, marker_value)| {
+                    !matches!(marker_value, MarkerValue::ConfigUpdate(_))
+                        || !already_written_config_updates
+                            .pop()
+                            .expect("Information on all config updates are present by `multi_contains_keys` above")
```

Comment:
Are we just asserting that these two vectors line up properly? 
Should we just use a map instead?

Reply:
Essentially, although a bit more complex -- we are asserting that the two vectors over the subset of `ConfigUpdate` marker values line up.

Note that the `markers` here are all markers to write (shared deleted, received, etc) and we don't want to not write those out or call `multi_contains_keys` for those so we only perform this query for config updates, and then as we see them we make sure they're in-synch. I might take a look at making this a bit nicer, as this is a bit hard to follow probably...